app/pods/components/utils/add-existing-document/component.js

```diff
@@ -1,76 +1,82 @@
-import Component from '@ember/component';
-import { inject } from '@ember/service';
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
 import { task, timeout } from 'ember-concurrency';
-import { computed, observer } from '@ember/object';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking'
 
+export default class AddExistingDocument extends Component {
+  @service store;
+  @tracked page = 0;
+  @tracked filter = '';
+  @tracked items = [];
 
-export default Component.extend({
+  size = 5;
+  sort = ['-created', 'name'];
 
-  store: inject(),
-
-  size: 5,
-  sort: '-created,name',
+  constructor() {
+    super(...arguments);
+    this.findAll.perform();
+  }
 
-  documents: computed('items.@each', function () {
-    (this.get('items') || []).map(item => item.set('selected', false));
-    return this.items;
-  }),
+  get pageParam() {
+    return this.page;
+  }
 
-  // dirty observers to make use of the datatable actions
-  pageObserver: observer('page', function () {
+  set pageParam(page) {
+    if (page === undefined) {
+      this.page = 0;
+    } else {
+      this.page = page;
+    }
     this.findAll.perform();
-  }),
+  }
+
+  setSelectedToFalse() {
+    this.items.map(item => item.set('selected', false));
+  }
 
-  queryOptions: computed('sort', 'filter', 'page', function () {
-    const { page, filter, size, sort } = this;
+  queryOptions() {
     let options = {
-      sort: sort,
+      sort: this.sort,
       page: {
-        number: page,
-        size: size
+        number: this.page,
+        size: this.size
       },
       filter: {},
     };
-    if (filter) {
-      options['filter']['name'] = filter;
+    if (this.filter) {
+      options.filter.name = this.filter;
     }
     return options;
-  }),
+  }
 
-  findAll: task(function* () {
-    const { queryOptions } = this;
-    const documents = yield this.store.query('document-version', queryOptions);
-    this.set('items', documents);
+  @task(function* () {
+    yield timeout(300);
+    const documents = yield this.store.query('document-version', this.queryOptions());
```

Comment:
`documents` variabele lijkt me hier overbodig, we doen er geen extra checks op dus zou rechtstreeks op items gezet worden.